File: app/api/schemas/audit.py
from typing import Any

from pydantic import BaseModel


class AuditEventSummary(BaseModel):
    event_type: str
    timestamp: str
    data: dict[str, Any]


class AuditTraceInfo(BaseModel):
    trace_id: str
    workflow: str
    started_at: str
    completed_at: str | None = None
    event_count: int
    events: list[AuditEventSummary] = []


class ListAuditTracesResponse(BaseModel):
    traces: list[AuditTraceInfo]
    total: int


class GetAuditTraceResponse(BaseModel):
    trace: AuditTraceInfo

File: types/api/audit.ts
export interface AuditEventSummary {
  event_type: string;
  timestamp: string;
  data: Record<string, unknown>;
}

export interface AuditTraceInfo {
  trace_id: string;
  workflow: string;
  started_at: string;
  completed_at: string | null;
  event_count: number;
  events: AuditEventSummary[];
}

export interface ListAuditTracesResponse {
  traces: AuditTraceInfo[];
  total: number;
}

export interface GetAuditTraceResponse {
  trace: AuditTraceInfo;
}
